load/dependencies.go

```diff
@@ -47,11 +47,18 @@ type Tracker[T comparable] interface {
 	// GetObservedFailed returns the number of transactions that the tracker has
 	// confirmed failed.
 	GetObservedFailed() uint64
+
+	// String returns a string representation of the tracker state.
+	String() string
+}
+
+type Logger interface {
```

Comment:
Lets not redefine the Logger interface here, and use the standard one from github.com/ava-labs/avalanchego/utils/logging

Generally, lets not proliferate interface definitions across packages just to try to slim down a few functions. It can lead to confusion about where the true interface definition lives, what functions should be available for future code changes, and cause more code changes in the future if things change.